lib/src/system_shell.dart

```diff
@@ -2,13 +2,26 @@ import 'dart:io';
 
 import 'package:path/path.dart' as path;
 
-/// The supported shells.
+/// {@template system_shell}
+/// A type definition for a shell.
+///
+/// A shell shell is a computer program that exposes an operating system's
+/// services.
```

Comment:
This could be more descriptive. To be more precise, it isn't up to this doc to explain what a shell is.